clients/src/main/java/org/apache/kafka/clients/consumer/StickyAssignor.java

```diff
@@ -705,6 +705,8 @@ static ByteBuffer serializeTopicPartitionAssignment(List<TopicPartition> partiti
      */
     private <T> boolean hasIdenticalListElements(Collection<List<T>> col) {
         Iterator<List<T>> it = col.iterator();
+        if (!it.hasNext())
+            return true;
```

Comment:
Could you please also add a unit test that covers this scenario? If I'm not mistaken `col` will be empty when either the list of partitions or the list of consumers passes to the assigner is empty.